ledger/tracker.go

```diff
@@ -282,6 +282,8 @@ func (dcc deferredCommitContext) newBase() basics.Round {
 var errMissingAccountUpdateTracker = errors.New("initializeTrackerCaches : called without a valid accounts update tracker")
 
 func (tr *trackerRegistry) initialize(l ledgerForTracker, trackers []ledgerTracker, cfg config.Local) (err error) {
+	tr.mu.Lock()
+	defer tr.mu.Unlock()
```

Comment:
it is init and single threaded

Reply:
My puzzle is, why do we have a `RLock` for `tr.dbRound` in `func (tr *trackerRegistry) loadFromDisk`, but not here?
Both `initialize` and `loadFromDisk` are called back to back from `reloadLedger`, so if one is single threaded, so should be the other one.

Reply:
right, reloadLedger is catchpoints is the answer.

Reply:
nit: I think we discussed this should be only taken to protect `tr.dbRound` and released right after.

Reply:
I know that we need a lock here if we need a lock in `loadFromDisk`, since both are called from the same location sequentially. 

I am not yet certain if, for example, `commitRound` can be called concurrently with `reloadLedger`. If the answer is 'No', perhaps we don't need the locks here. 

If the answer is 'Yes', I think some other things need to be taken care of. For example, `tr.trackers = nil` will run concurrently with `range tr.trackers`.

Moreover, a lock will be needed for `tr.lastFlushTime` in `relplay`, since it will interfere with `tr.lastFlushTime = dcc.flushTime` in `commitRound`.

Finally, `tr.accountsWriting` may need more attention. As well as `tr.deferredCommits` will be created new in `initialize` concurrently with `tr.deferredCommits <- dcc` in `scheduleCommit`.

Reply:
This will be addressed as part of https://github.com/algorand/go-algorand-internal/issues/3021